mpp/mpp_efp.F90

```diff
@@ -257,8 +257,8 @@ function mpp_reproducing_sum_r4_2d(array, isr, ier, jsr, jer, EFP_sum, reproduci
 
   array_r8 = array
 
-  sum = mpp_reproducing_sum_r8_2d(array_r8, isr, ier, jsr, jer, EFP_sum, reproducing, &
-                            overflow_check, err)
+  sum = real(mpp_reproducing_sum_r8_2d(array_r8, isr, ier, jsr, jer, EFP_sum, reproducing, &
```

Comment:
This needs extensive testing.

Reply:
The return from the mpp_reproducing_sum is a real(kind=8) and sum is defined as a real(kind=4).  With this addition, the cast is no longer implicit.

Reply:
Will removing the implicit cast change answers/rounding though? Is the behavior of the explicit cast the same as the behavior of the implicit cast?  Can we guarantee the behavior between compilers?  Is it part of the standard? 
I have a repressed memory of @menzel-gfdl and I "fixing" something similar to this involving `mpp_reproducing_sum`, and it changed answers when variables were being explicitly cast.